app/src/main/java/ch/epfl/sdp/lobby/game/GameLobbyActivity.kt

```diff
@@ -99,20 +102,24 @@ class GameLobbyActivity : AppCompatActivity(), SwipeRefreshLayout.OnRefreshListe
 
     override fun onFactionChange(newFaction: Faction) {
         //player id is hardcoded for now
-        repository.setPlayerFaction(gameID, userID, newFaction)
-        myFaction = newFaction
-        updateLocalPlayerState()
+        repository.setPlayerFaction(gameID, userID, newFaction) {
+            myFaction = newFaction
+            updateLocalPlayerState()
+        }
     }
 
     private fun updateLocalPlayerState() {
+        var newReadyState by Delegates.notNull<Boolean>()
         if(myFaction == Faction.PREY && myTag == null) {
             gameLobbyBinding.txtPlayerReady.text = getString(R.string.you_are_not_ready)
-            repository.setPlayerReady(gameID, userID, false)
+            newReadyState = false
         } else {
             gameLobbyBinding.txtPlayerReady.text = getString(R.string.you_are_ready)
-            repository.setPlayerReady(gameID, userID, true)
+            newReadyState = true
+        }
```

Comment:
Maybe using an if-expression would be clearer here than a delegate :
val newReadyState = if(...) {
        gameLobbyBinding.txtPlayerReady...
        false
    else {
        gameLobbyBinding.txtPlayerReady...
        true
    }